psh/runfile/runfile.c

```diff
@@ -80,7 +80,20 @@ int psh_runfile(int argc, char **argv)
 	/* Take back terminal control */
 	tcsetpgrp(STDIN_FILENO, getpgid(getpid()));
 
-	return ((pid > 0) && (ret >= 0)) ? WEXITSTATUS(status) : EXIT_FAILURE;
+	/* Determine the status of fhe file */
+	if (!((pid > 0) && (ret >= 0))) {
+		return_value = EXIT_FAILURE;
+	}
+	if (0 != WIFSIGNALED(status)) {
```

Comment:
I get that there are some rationale about `const != variable`, but:
- it doesn't really apply to `!=` operator, because `==` is a "dangerous" one,
- gcc emits warning when it sees `if (var = const)`
- it's inconsistent with our coding style.
```suggestion
	else if (WIFSIGNALED(status) != 0) {
```